app/screens/integration_selector/index.tsx

```diff
@@ -0,0 +1,634 @@
+
+        if (receivedChannels) {
+            setSearchResults(receivedChannels);
+        }
+    };
+
+    const searchProfiles = async (searchTerm: string) => {
+        const {data: userData} = await searchProfilesRemote(serverUrl, searchTerm.toLowerCase(), {team_id: currentTeamId, allow_inactive: true});
+
+        if (userData) {
+            setSearchResults(userData);
+        }
+    };
+
+    const searchDynamicOptions = useCallback((searchTerm = '') => {
+        if (options) {
+            setIntegrationData(options);
+        }
+
+        if (!getDynamicOptions) {
+            return;
+        }
+
+        getDynamicOptions(searchTerm.toLowerCase()).then((results: DialogOption[]) => {
+            const searchData = results || [];
+
+            if (searchTerm) {
+                setSearchResults(searchData);
+            } else {
+                setIntegrationData(searchData);
+            }
+        });
+    }, [term, integrationData, searchResults]);
+
+    const onHandleMultiselectSubmit = () => {
+        handleSelectItem(getMultiselectData());
+        close();
+    };
+
+    const onSearch = useCallback((text: string) => {
+        if (!text) {
+            clearSearch();
+        }
+
+        setTerm(text);
+
+        if (searchTimeoutId.current) {
+            clearTimeout(searchTimeoutId.current);
+        }
+
+        searchTimeoutId.current = setTimeout(async () => {
+            if (!dataSource) {
+                setSearchResults(filterSearchData('', integrationData, text));
+                return;
+            }
+
+            setLoading(true);
+
+            if (dataSource === ViewConstants.DATA_SOURCE_USERS) {
+                await searchProfiles(text);
+            } else if (dataSource === ViewConstants.DATA_SOURCE_CHANNELS) {
+                await searchChannels(text);
+            } else if (dataSource === ViewConstants.DATA_SOURCE_DYNAMIC) {
+                await searchDynamicOptions(text);
+            }
+
+            setLoading(false);
+        }, General.SEARCH_TIMEOUT_MILLISECONDS);
+    }, [dataSource, term, searchResults]);
+
+    const filterSearchData = (source: string, searchData: DataType, searchTerm: string) => {
+        if (!data) {
+            return [];
+        }
+
+        const lowerCasedTerm = searchTerm.toLowerCase();
+
+        if (source === ViewConstants.DATA_SOURCE_USERS) {
+            return filterProfilesMatchingTerm(searchData as UserProfile[], lowerCasedTerm);
+        } else if (source === ViewConstants.DATA_SOURCE_CHANNELS) {
+            return filterChannelsMatchingTerm(searchData as Channel[], lowerCasedTerm);
+        } else if (source === ViewConstants.DATA_SOURCE_DYNAMIC) {
+            return searchData;
+        }
+
+        return (searchData as DialogOption[]).filter((option) => option.text && option.text.toLowerCase().startsWith(lowerCasedTerm));
+    };
+
+    const getMultiselectData = useCallback((): Selection => {
```

Comment:
In theory this depends on the `dataSource` (I guess it is constant during the lifetime of this component, though)